S2VX.Game/Story/Note/GameNote.cs

```diff
@@ -53,17 +53,19 @@ public class GameNote : S2VXNote, IKeyBindingHandler<InputAction> {
             if (Story.Notes.HasClickedNote) {
                 return false;
             }
-
             var time = Time.Current;
-            // Limit timing error to be +/- MissThreshold (though it will never be >= MissThreshold since RecordMiss would have already run)
-            TimingError = (int)Math.Round(Math.Clamp(time - HitTime, -MissThreshold, MissThreshold));
             var isVisible = Alpha > 0;
-            Story.Notes.HasClickedNote = true;
-            return isVisible;
+            var isWithinMissThreshold = Math.Abs(time - HitTime) <= MissThreshold;
+            if (!isVisible || !isWithinMissThreshold) {
+                return false;
+            }
+            TimingError = (int)Math.Round(time - HitTime);
```

Comment:
```suggestion
            TimingError = time - HitTime;
```

One change I made was to make score a `double` for accuracy. The `ScoreInfo` class handles rounding to an `int`. Can we remove the rounding stuff here and see what happens?

Reply:
Should I change MissThreshold to be a double as well?

Reply:
Yeah, I think that would make sense and calculations easier.

Reply:
Alright I'll change it for GameNote and GameHoldNote but still pass in int to the ScoreBar since we're gonna get rid of it later anyways and I don't want to refactor it in this change.